Assets/Plugins/LeapMotion/Modules/Hands/Scripts/HandBinder/HandBinderAutoRigger.cs

```diff
@@ -0,0 +1,217 @@
+        /// <returns></returns>
+        private static Transform[] SortBones(Transform[] bones, bool isThumb = false) {
+            Transform meta = null;
+            Transform proximal = null;
+            Transform middle = null;
+            Transform distal = null;
+
+            if(isThumb || bones.Length == 3) {
+                meta = null;
+                proximal = bones[0];
+                middle = bones[1];
+                distal = bones[2];
+            }
+            //We assume the 4th child is the distal bone
+            else if(bones.Length >= 4) {
+                meta = bones[0];
+                proximal = bones[1];
+                middle = bones[2];
+                distal = bones[3];
+            }
+
+            var boundObjects = new Transform[]
+            {
+                meta,
+                proximal,
+                middle,
+                distal
+            };
+
+            return boundObjects;
+        }
+
+        /// <summary>
+        /// Bind a transform in the scene to the Hand Binder
+        /// </summary>
+        /// <param name="boneTransform">The transform you want to assign </param>
+        /// <param name="fingerIndex"> The index of the finger you want to assign</param>
+        /// <param name="boneIndex">The index of the bone you want to assign</param>
+        /// <param name="handBinder">The Hand Binder this information will be added to</param>
+        /// <returns></returns>
+        public static BoundBone[] AssignUnityBone(Transform[] boneTransform) {
+            var boundFingers = new BoundBone[]
+                {
+                    AssignBoundBone(boneTransform[0]),
+                    AssignBoundBone(boneTransform[1]),
+                    AssignBoundBone(boneTransform[2]),
+                    AssignBoundBone(boneTransform[3]),
+                };
+
+            return boundFingers;
+        }
+
+        public static BoundBone AssignBoundBone(Transform transform) {
+            var newBone = new BoundBone();
+            if(transform != null) {
+                newBone.boundTransform = transform;
+                newBone.startTransform = new TransformStore();
+                newBone.startTransform.position = transform.localPosition;
+                newBone.startTransform.rotation = transform.localRotation.eulerAngles;
+            }
+            return newBone;
+        }
+
+        /// <summary>
+        /// Calculate the rotation offset needed to get the rigged hand into the same orientation as the leap hand
+        /// </summary>
+        public static void CalculateWristRotationOffset(HandBinder handBinder) {
+            Transform middleProximal = handBinder.boundHand.fingers[2].boundBones[1].boundTransform;
+            Transform indexProximal = handBinder.boundHand.fingers[1].boundBones[1].boundTransform;
+            Transform pinkyProximal = handBinder.boundHand.fingers[4].boundBones[1].boundTransform;
```

Comment:
It seems like the wrist rotation misalignment I notice can be fixed by changing the `boundBones` indices from `[1]` to `[0]`.

This is an example of the change:
![BadToGoodWristOrientation](https://user-images.githubusercontent.com/174475/106528213-42e6d800-649d-11eb-8cd6-faacacb15449.gif)

Reply:
1.) Will do, I was using 2019.4 instead of 2019.2 as the UI looks so gross, look at the difference between the custom editor.
![image](https://user-images.githubusercontent.com/75795822/106582742-3bc2d700-653c-11eb-9927-897e075b0892.png)

2.) Great spot!
- So when pressing the auto rig button, the hand gets reset to its initial model pose, this is to stop re-calculating the rotation difference from the hand that is in the editor pose. 
- Then once it is  in its default pose the rotation offset is calculated and scene GUI update is triggered but the call to actually update the hand to its new pose was happening to late. So by pressing the Autorig button on the other hand, it was finally updating the hand to the new rotation offsets that were calculated the time before.

This has been fixed by simply updating the hand at the end of the AutoRig function rather than waiting for scene GUI to get updated.

3.) While I agree this does fix it, some models do not include meta finger bones which is what indices [0] is. So, as a fix I will change the code so that it tries to use indices [0] first then if these do not exist I will then use the proximal bones.

4.) Another awesome spot, I will make sure this also goes into the updated documentation that I have written so that our users are aware of the difference between pivots at the end of the previous bone vs the start of the next bone. 

https://gyazo.com/89338609e7399d84d804c40329c67dec - This is what it is like currently.
https://gyazo.com/91f558c815848b6d1425e50379ab21ba - After your suggestion

Reply:
1. The recommendation from Unity is to keep `packages-lock.json` under [source control](https://docs.unity3d.com/Manual/upm-conflicts-auto.html) such that the same versions are pulled down by all users. It probably doesn't really matter for the case of packages managed by Unity directly as everything uses semantic versioning so there should not be a situation where a package or its dependencies change, but it is relevant in the case where git repositories are added since it uses a specific commit hash in the lock file to mark the required version.

Are `"com.unity.xr.oculus.standalone"` and `"com.unity.xr.openvr.standalone"` dependencies that have been intentionally added? If so I guess the best course of action would be to revert `manifest.json` entirely to get back to the base dependencies and undo any automatic package updates between 2019.2 and 2019.4, then optionally add `""com.unity.xr.legacyinputhelpers"` (instead of `"com.unity.xr.arfoundation"`). I don't think things like `"com.unity.ide.vscode"` should be in there by default though unless we add in adapters for all relevant IDEs.